pkgs/development/python-modules/python-miio/default.nix

```
@@ -22,10 +24,20 @@ buildPythonPackage rec {
     sha256 = "3be5275b569844dfa267c80a1e23dc0957411dd501cae0ed3cccf43467031ceb";
   };
 
+  postPatch = ''
+    # use any available version from nixpkgs
+    substituteInPlace setup.py \
+      --replace "pytz>=2019.3,<2020.0" "pytz" \
+      --replace "cryptography>=2.9,<3.0" "cryptography" \
+      --replace "zeroconf>=0.25.1,<0.26.0" "zeroconf"
+  '';
+
   checkInputs = [ pytest ];
-  propagatedBuildInputs = [ appdirs click construct cryptography zeroconf attrs pytz tqdm netifaces ];
+  propagatedBuildInputs = [ appdirs click construct cryptography zeroconf attrs pytz tqdm netifaces croniter importlib-metadata ];
 
   checkPhase = ''
+    # tests/test_device.py appears to be broken
```

comments within code blocks can cause rebuilds if changed, preference to have them as nix comments
```suggestion
  # tests/test_device.py appears to be broken
  checkPhase = ''
```